src/org/intellij/erlang/compilation/ErlangPrepareDependenciesCompileTask.java

```diff
@@ -125,164 +127,150 @@ private static ErlangModuleBuildOrders getModuleBuildOrders(CompileContext conte
-          moduleNames.addAll(ErlangPsiImplUtil.getImplementedBehaviourModuleNames(module.getModuleFile()));
-          List<Node> dependencies = getModuleNodes(moduleNames);
-          module.addDependencies(dependencies);
-          module.addDependencies(globalPtNodes);
-        }
-      }
-
-      private List<Node> getModuleNodes(Collection<String> nodesName) {
-        return ContainerUtil.mapNotNull(nodesName, new Function<String, Node>() {
-          @Override
-          public Node fun(String pt) {
-            return myNamesToNodesMap.get(pt);
-          }
-        });
-      }
+    @NotNull
+    private List<String> getDeclaredParseTransformPaths(@NotNull ErlangFile erlangModule) {
+      Set<String> pt = ContainerUtil.newHashSet();
+      erlangModule.addDeclaredParseTransforms(pt);
+      return getPathsFromNames(pt);
     }
 
-    private static class Node {
-      private final ErlangFile myModuleFile;
-      private final List<Node> myDependencies = ContainerUtil.newArrayList();
-
-      Node(ErlangFile moduleFile) {
-        myModuleFile = moduleFile;
-      }
-
-      @Override
-      public int hashCode() {
-        return myModuleFile.getName().hashCode();
-      }
-
-      @Override
-      public boolean equals(Object o) {
-        if (this == o) return true;
-        if (o == null || getClass() != o.getClass()) return false;
-
-        Node node = (Node) o;
-
-        return myModuleFile.getName().equals(node.myModuleFile.getName());
+    private void buildNamesMap(@NotNull Collection<VirtualFile> erlangModules) {
+      for (VirtualFile erlangModule : erlangModules) {
+        myNamesToPathsMap.put(erlangModule.getNameWithoutExtension(), erlangModule.getPath());
       }
+    }
 
-      public void addDependencies(@NotNull Collection<Node> deps) {
-        for (Node dep : deps) {
-          addDependency(dep);
+    @NotNull
+    private List<String> getPathsFromNames(@NotNull Collection<String> erlangModuleNames) {
+      return ContainerUtil.mapNotNull(erlangModuleNames, new Function<String, String>() {
+        @Override
+        public String fun(String name) {
+          return myNamesToPathsMap.get(name);
         }
-      }
-
-      ErlangFile getModuleFile() {
-        return myModuleFile;
-      }
+      });
+    }
 
-      String getModuleName() {
-        return FileUtil.getNameWithoutExtension(myModuleFile.getName());
-      }
+    @Override
+    public Collection<String> getNodes() {
+      return myPathsToDependenciesMap.keySet();
+    }
 
-      void addDependency(@Nullable Node dep) {
-        if (dep != null && dep != this) {
-          myDependencies.add(dep);
-        }
-      }
+    @Override
+    public Iterator<String> getIn(String filePath) {
+      return myPathsToDependenciesMap.get(filePath).iterator();
+    }
 
-      @NotNull
-      List<Node> getDependencies() {
-        return myDependencies;
-      }
+    @Nullable
+    public List<String> getDependencies(@NotNull String filePath) {
```

Comment:
You've declared this method as nullable, but only use it as not null - why not make it not-null instead?